pkg/new-ui/v1beta1/frontend/src/app/shared/params-list/types.ts

```diff
@@ -0,0 +1,15 @@
+export type ParameterType = 'int' | 'double' | 'discrete' | 'categorical';
+
+export interface Range {
+  min: string;
+  max: string;
+  step: string;
+}
+
+export type FeasibleSpace = Range | any[];
+
+export interface ParameterSpec {
```

Comment:
We can't re-use interface ParameterSpec from the models: https://github.com/kubeflow/katib/blob/1b48a137ac24428d83bd411bc1ce4d3f4724d1c6/pkg/new-ui/v1beta1/frontend/src/app/models/experiment.k8s.model.ts#L57 ?

Reply:
Indeed it makes sense to do so. There are only some slight changes I need to make but this should be the way to go, since the types defined in the `types.ts` file are almost the same with the ones in the `models` file.
On it

Reply:
This turned out to be a little bit more involved than I anticipated, but I think it really helped simplify some of the form's logic.

I was using an intermediate json variables for the form value that were not 100% aligned with the naming of the CR's fields. For example the form's json representation [ values of the [FormControls](https://angular.io/guide/reactive-forms#creating-nested-form-groups) ] for hyper parameters was:
```json
{
  "name": "some-name",
  "type": "int",
  "value": {
    "min": "1",
    "max": "64"
  }
}
```

while the `ParameterSpec` type is
```json
{
  "name": "some-name",
  "parameterType": "int",
  "feasibleSpace": {
    "min": "1",
    "max": "64"
  }
}
```

I pushed a commit that updates the form's JSON representation, via the FormControls' values, to be the same with the actual `ParameterSpec` interface. This will make it slightly easier to keep track of what's going on since the form will have the same structure as the CR's API.

While it was not a trivial change, it was definitely one in the right direction!

Reply:
Lastly, the form's submitted data looks OK after this change. Also notice that the `step` is omitted for `--lr` and the estimation of created Trials is not visible, since the step is omitted

[ the purple-ish color is because I've messed up my laptop's color settings ]

![Screenshot from 2021-02-26 01-06-15](https://user-images.githubusercontent.com/11134742/109231309-02922580-77cf-11eb-8b99-d8fe2b38ad03.png)

Reply:
> I pushed a commit that updates the form's JSON representation, via the FormControls' values, to be the same with the actual ParameterSpec interface. This will make it slightly easier to keep track of what's going on since the form will have the same structure as the CR's API.

That's great, thank you! What about other parameters, does it also require some changes between values for the FormControl and API representation ? Or this problem was only for the `ParameterSpec` ?

Reply:
While the other FormControl API representations are not 1-1 with the CR's fields I think it's OK. The differences are mostly helper controls, like `useStrategies` for the checkbox around setting metric strategies, an extra control to differentiate the algorithm type [ nas, hp ] etc.

Reply:
Resolving the conversation. You know the drill, if you need us to further discuss feel free to re-open it